### ISSUE TITLE
Snackbar Accessibility Issues

### ISSUE THREAD

**Issue by swissspidy** [GoogleForCreators/web-stories-wp#6061]
## Feature Description

Our current snackbars are not really accessible, as they are not announced to screen readers, and it's impossible for users to interact with them.

I know that there are general issues with snackbars (bad location, auto-disappearance, etc.), but let's do one step at a time.

## Additional Context

Recommended reading including documents linked to: https://github.com/WordPress/gutenberg/issues/16549

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria

Make snackbars more accessible

## Implementation Brief

* [ ] Snackbar is announced to screen readers (#6059)
* [ ] If the snackbar contains an action, there is a global shortcut to perform said action

**Comment by swissspidy** [comment]
Oh, and we can then probably drop the ARIA label for the snackbar messages once we use the live region

**Comment by BrittanyIRL** [comment]
@miina - can i grab this from you?

**Comment by miina** [comment]
@BrittanyIRL Sure!

**Comment by samwhale** [comment]
After some research - I believe that snackbars _should not_ steal focus from whatever has the current focus. It's not an [alert dialog](https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/alertdialog.html), and snacks are supposed to be "minimally disruptive". They don't require attention, so forcing the user to interact with them when one shows up seems disruptive.

![image.png](https://images.zenhubusercontent.com/5b1032a1348b7966b43421a4/f7cd7843-b2f6-48ea-acd1-4f09762b3199)
https://material.io/components/snackbars#usage

But if the snackbar were to steal focus, we'd also need a way to return focus to the previously selected element. This could be done with portals I guess, but it could be confusing to users about where the focus were to go if `Tab` were ever pressed.

I think the messages should be announced to the screen reader, and a global shortcut should be registered to interact with the action in the snack. The message should communicate the shortcut to the user so that they may perform the action with the keyboard. Thoughts @miina @swissspidy @BrittanyIRL ?

**Comment by swissspidy** [comment]
That matches the suggestion in the ticket description, so +1 from me.

**Comment by samwhale** [comment]
Oh perfect 😅 I thought it was going to steal focus for some reason, but re-reading the ticket description I now realize I was mistaken. Thanks!

**Comment by samwhale** [comment]
I don't think this needs the `Status: Needs More Info` tag anymore

**Comment by csossi** [comment]
Verified in QA
